test/integration.sh

```diff
@@ -1,47 +1,72 @@
 
-    [[ -d outputs ]] && sudo rm -rf outputs && INFO "deleted old outputs"
+    # Get array of all required containers from file.
+    fname=expected_containers
+    IFS=$'\r\n' GLOBIGNORE='*' command eval  'containers_expected=($(cat $fname))'
+    containers_expected=($containers_expected)  # cast as array
 
-    access_token=$(../redwood/cli/bin/redwood token create)
-    INFO "generated testing access token: ${access_token}"
+    num_containers_exp="${#containers_expected[@]}"
 
-    redwood_endpoint=$(cat ../redwood/.env | grep 'base_url=' | sed 's/[^=]*=//')
-    INFO "got redwood endpoint from dcc-ops/redwood/.env: ${redwood_endpoint}"
+    # Crude initial test to verify that at least 7 containers are running.
+    [ "${#containers[@]}" -ge $num_containers_exp ] ||\
+        (ERROR "Not running the required number of docker containers." && exit 1)
 
-    INFO "doing upload with $(pwd)/manifest.tsv"
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/manifest.tsv:/dcc/manifest.tsv -v $(pwd)/samples:/samples -v $(pwd)/outputs:/outputs \
-         quay.io/ucsc_cgl/core-client:1.1.0-alpha spinnaker-upload --skip-submit --force-upload /dcc/manifest.tsv
+    for container_expected in "${containers_expected[@]}"; do
+        counter=0
+        for container in "${containers[@]}"; do
+            if [[ $container_expected = $container ]]; then
+                continue
+            else
+                counter=$((counter+1))
+            fi
+        done
 
-    object_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 20)
-    bundle_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 19)
-    INFO "the uploaded bundle with id ${bundle_id} has metadata.json with object id ${object_id}"
+        if [[ "$counter" -eq $num_containers_exp ]]; then
+            ERROR "One or more docker containers not running." && exit 1
+        fi
+    done
 
-    INFO "downloading metadata.json with object id ${object_id}"
-    sudo mkdir outputs/test_download
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/outputs:/outputs quay.io/ucsc_cgl/core-client:1.1.0-alpha download ${object_id} /outputs/test_download
+}
+
+function main {
+    # This tests whether all Docker containers are installed and then
+    # hits the DCC Dashboard Host and checks whether it returns a 200
+    # status code.
+
+    cd "$(dirname "${BASH_SOURCE[0]}")"
+    LOG_LEVEL_ALL
+    check_setup
 
-    [[ ! -f outputs/test_download/${bundle_id}/metadata.json ]] && printf "%s: no such file" "outputs/test_download/${bundle_id}/metadata.json" | FATAL && exit 1
+    # Get name of the DCC Dashboard Host.
+    curr_dir=$(pwd)
+    cd ../boardwalk
+    dcc_host=$(grep -o 'dcc_dashboard_service.*' .env | cut -f2- -d=)
+    cd "$curr_dir"
 
-    sudo rm -rf outputs
+    url="https://$dcc_host/api/v1/repository/files/export?/"
```

Comment:
You shouldn't need `?`. You shouldn't need trailing slash either, although I think it does a redirect, so I can see why you added it. If you want to remove it, you can change curl to follow redirects.

Reply:
Removed `?` and the slash and used the follow-redirects option for `cURL`.